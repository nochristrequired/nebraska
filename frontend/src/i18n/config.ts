import i18next from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import { initReactI18next } from 'react-i18next';

const en = {}; // To keep TS happy.

i18next
  // detect user language https://github.com/i18next/i18next-browser-languageDetector
  .use(LanguageDetector)
  .use(initReactI18next)
  // Use dynamic imports (webpack code splitting) to load javascript bundles.
  // @see https://www.i18next.com/misc/creating-own-plugins#backend
  // @see https://webpack.js.org/guides/code-splitting/
  .use({
    type: 'backend',
    read<Namespace extends keyof typeof en>(
      language: string | any,
      namespace: Namespace,
      callback: (errorValue: unknown, translations: null | typeof en[Namespace]) => void
    ) {
      import(`./locales/${language}/${namespace}.json`)
        .then(resources => {
          callback(null, resources);
        })
        .catch(error => {
          callback(error, null);
        });
    },
  })
  // i18next options: https://www.i18next.com/overview/configuration-options
  .init({
    debug: process.env.NODE_ENV === 'development',
    fallbackLng: 'en',
    supportedLngs: ['en'],
    // nonExplicitSupportedLngs: true,
    interpolation: {
      escapeValue: false, // not needed for react as it escapes by default
      format: function (value, format, lng) {
        // https://www.i18next.com/translation-function/formatting
        if (format === 'number') return new Intl.NumberFormat(lng).format(value);
        if (format === 'date')
          return new Intl.DateTimeFormat(lng, {
            day: '2-digit',
            month: '2-digit',
            year: 'numeric',
          }).format(value);

        return value;
      },
    },
    returnEmptyString: false,
    // https://react.i18next.com/latest/i18next-instance
    // https://www.i18next.com/overview/configuration-options
    react: {
      useSuspense: false, // not needed unless loading from public/locales
      //   bindI18nStore: 'added'
    },
    nsSeparator: '|',
    keySeparator: false,
  });

export default i18next;
