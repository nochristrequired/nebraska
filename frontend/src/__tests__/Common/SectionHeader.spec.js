import '../../i18n/config.ts';
import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import SectionHeader from '../../components/Common/SectionHeader';

describe('Section Header', () => {
  const minProps = {
    breadcrumbs: [
      {
        path: '/apps',
        label: 'Applications',
      },
    ],
    title: 'Flatcar',
  };
  it('renders section header correctly', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <SectionHeader {...minProps} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
