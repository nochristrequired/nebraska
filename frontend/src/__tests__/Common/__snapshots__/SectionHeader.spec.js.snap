// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Header renders section header correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-sectionContainer-1 MuiGrid-container MuiGrid-align-items-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <nav
        aria-label="breadcrumbs"
        class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        <ol
          class="MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <span
              class="MuiBox-root MuiBox-root-3 makeStyles-breadCrumbsItem-2"
            >
              <a
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
                href="/apps"
              >
                Applications
              </a>
            </span>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
            >
              Flatcar
            </p>
          </li>
        </ol>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
